Packages/Core/Runtime/Conditions/AtomCondition.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using UnityEngine;
+
+namespace UnityAtoms
+{
+    /// <summary>
+    /// Base abstract class for Conditions. Inherits from `AtomFunction` and 
+    /// must implement `Condition` Predicate
+    /// </summary>
+    public abstract class AtomCondition<T> : AtomFunction<T, T>
+    {
+        public Predicate<T> Predicate;
```

Comment:
Should actually be `AtomCondition<T> : AtomFunction<bool, T>`
and since AtomFunction already defines a `public Func` this is already in existence here, so the Predicate<T> delegate is not needed.